lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -372,32 +372,57 @@ def add_test_targets
             end
           end
 
+          # Attaches an app host target to a given test target
+          #
+          # @param [PBXNativeTarget] app_host_target
+          # @param [PBXNativeTarget] test_native_target
+          #
+          def attach_app_host_target_to_test_native_target(app_host_target, test_native_target)
+            test_native_target.build_configurations.each do |configuration|
+              test_host = "$(BUILT_PRODUCTS_DIR)/#{app_host_target.name}.app/"
+              test_host << 'Contents/MacOS/' if target.platform == :osx
+              test_host << app_host_target.name.to_s
+              configuration.build_settings['BUNDLE_LOADER'] = '$(TEST_HOST)'
```

Comment:
Just because I'm curious, why is the `BUNDLE_LOADER` build setting necessary?

Reply:
I'm genuinely not sure. Even without these changes, auto-generated app hosts wouldn't render correctly as the app host in xcode. This fixes that.

Reply:
Was it saying "Custom" or "Unknown"? 

In the future if we add XCUITest support this setting actually should _not_ be set. You can catch this from now by passing `test_type` which you can read from the consumer with the value of `:unit`.